RUFAS/biophysical/field/crop/nutrient_uptake.py

```
@@ -10,13 +9,13 @@ class NutrientUptake(ABC):
     ----------
-    crop_data : Optional[CropData], optional
-        An instance of `CropData` containing crop specifications and attributes.
-        Defaults to a new instance of `CropData` if not provided.
+    crop_data : CropData, optional
+        An instance of ``CropData`` containing crop specifications and attributes.
+        Defaults to a new instance of ``CropData`` if not provided.
 
     """
 
-    def __init__(self, crop_data: Optional[CropData]):
+    def __init__(self, crop_data: CropData | None):
```

Ruff (0.15.15)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)